zip-seanbowe-esktransfer.rst

```diff
@@ -0,0 +1,70 @@
+
+Terminology
+===========
+
+The key words "MUST" and "MUST NOT" in this document are to be interpreted as described in RFC 2119.
+
+Abstract
+========
+
+This ZIP proposes a change to the transaction structure to accommodate a new field in the note plaintext transmitted to recipients of Sapling payments, the purpose of which is to avoid an assumption on zk-SNARK knowledge soundness for preventing diversified address linkability.
+
+Motivation
+==========
+
+The Sapling payment protocol contains a feature called "diversified addresses" which allows a single incoming viewing key to receive payments on an enormous number of distinct and unlinkable payment addresses. This feature allows users to maintain many payment addresses without paying additional overhead during blockchain scanning.
+
+The feature works by allowing payment addresses to become a tuple (``pk_d``, ``d``) of a public key ``pk_d`` and 88-bit diversifier ``d`` such that ``pk_d = [ivk] GH(d)`` for some incoming viewing key ``ivk``. The hash function ``GH(d)`` maps from a diversifier to prime order elements of the Jubjub elliptic curve. It is possible for a user to choose many ``d`` to create several distinct and unlinkable payment addresses of this form.
+
+In order to make a payment to a Sapling address, an ephemeral secret ``esk`` is sampled by the sender and an ephemeral public key ``epk = [esk] GH(d)`` is included in the Output description. Then, a shared Diffie-Hellman secret is computed by the sender as ``[esk] [8] pk_d``. The recipient can recover this shared secret without knowledge of the particular ``d`` by computing ``[ivk] [8] epk``. This shared secret is then used as part of Note decryption.
+
+Naively, the recipient cannot know which ``(pk_d, d)`` was used to compute the shared secret, but the sender is asked to include the ``d`` within the note plaintext to reconstruct the Note. However, if the recipient has many known addresses, an attacker could use a different payment address to perform secret exchange and, by observing the behavior of the recipient, link the two diversified addresses together.
+
+In order to prevent this attack, the protocol currently forces the sender to prove knowledge of the discrete logarithm of ``epk`` with respect to the ``g_d`` included within the Note commitment. This ``g_d`` is determined by ``d`` and recomputed during Note decryption, and so the recipient will either be unable to decrypt the note or the sender will be unable to perform the attack.
+
+However, this check occurs as part of the zero-knowledge proof statement and so relies on the knowledge soundness of the underlying zk-SNARK in Sapling. It would be preferable to force the sender to transfer ``esk`` within the Note plaintext so that, during note decryption, the recipient can ensure that it corresponds with ``epk`` and ``d`` correctly and ignore the payment as invalid otherwise. This allows a line of defense in the case that knowledge soundness of the zk-SNARK does not hold.
```

Comment:
Instead of "corresponds with" it would help to specify exactly what the recipient checks.